### PR TITLE
Update equals method for UniqueStudentList

### PR DESCRIPTION
**Pull request by JonahhGohh**
1. Update equals method for UniqueStudentList so that when testing for equality of model and addressbook also checks for the equality between sessions.
2. Update the typicalStudentAddressBook.json file to match that of TypicalStudents.
3. In DeleteSessionCommandTest success case, note that i avoided using getTypicalStudents() for the model and instead manually added them because I did not want the same student object in both models.

image
image
Now the student and sessions are unique. whereas previously when calling getTypicalAddressBook() all students and sessions are  the same objects.

### CODE REVIEW

File: src/test/java/seedu/address/logic/commands/DeleteSessionCommandTest.java [hunk 1]
```diff
@@ -24,25 +30,50 @@
 
     private Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
 
-    /*
+
     @Test
-    public void execute_validNameValidIndex_success() {
-        Student studentWithSession = model.getAddressBook().getStudentList().get(INDEX_FIRST_STUDENT.getZeroBased());
+    public void execute_validNameValidIndex_success() throws CommandException {
+        AddressBook deletionAddressBook = new AddressBook();
```
Comment (nowknowing) [inline src/test/java/seedu/address/logic/commands/DeleteSessionCommandTest.java:36]: we put this outside the method, at the top of the class?
to use in every other method for test

**Review by nowknowing — APPROVED** [review]
LGTM
whether or not theres a use for equals Student checking its sessions, i think this is fine

File: src/test/java/seedu/address/logic/commands/DeleteSessionCommandTest.java (hunk 1 above)
Comment (JonahhGohh) [reply src/test/java/seedu/address/logic/commands/DeleteSessionCommandTest.java:36]: Are we adding more tests to DeleteSessionCommandTest?

Comment (nowknowing) [reply src/test/java/seedu/address/logic/commands/DeleteSessionCommandTest.java:36]: Yea I think like failure cases and all that

Comment (nowknowing) [reply src/test/java/seedu/address/logic/commands/DeleteSessionCommandTest.java:36]: Improves coverage to do that right i hope

**Review by samleewy — APPROVED** [review]
sorry for the late review! everything looks good. there's still outstanding comments from @nowknowing which you may wanna address.

File: src/test/java/seedu/address/logic/commands/DeleteSessionCommandTest.java (hunk 1 above)
Comment (JonahhGohh) [reply src/test/java/seedu/address/logic/commands/DeleteSessionCommandTest.java:36]: hollup if we abstract this method and put it at the top of the class, then won't we be using the same edited addressbook for other test cases as well?

Comment (nowknowing) [reply src/test/java/seedu/address/logic/commands/DeleteSessionCommandTest.java:36]: oh right! so i guess it has to stay as it is rn

**Pull request merged**